kedro/framework/cli/catalog.py

```diff
@@ -250,35 +240,25 @@ def resolve_patterns(metadata: ProjectMetadata, env: str) -> None:
     explicit_datasets = {
         ds_name: ds_config
         for ds_name, ds_config in catalog_config.items()
-        if not data_catalog._is_pattern(ds_name)
+        if not data_catalog.config_resolver.is_pattern(ds_name)
     }
 
     target_pipelines = pipelines.keys()
-    datasets = set()
+    pipeline_datasets = set()
 
     for pipe in target_pipelines:
         pl_obj = pipelines.get(pipe)
         if pl_obj:
-            datasets.update(pl_obj.datasets())
+            pipeline_datasets.update(pl_obj.datasets())
 
-    for ds_name in datasets:
-        is_param = ds_name.startswith("params:") or ds_name == "parameters"
-        if ds_name in explicit_datasets or is_param:
+    for ds_name in pipeline_datasets:
```

Comment:
Not a big fan of `continue` since it harms the flow of thinking while reading the code. Could we do this instead:

```
for ds_name in (pipeline_datasets - explicit_datasets - filter(is_parameter, pipeline_datasets)):
```

Reply:
We will have to do this since `explicit_datasets` is a dictionary:

```
for ds_name in (pipeline_datasets - set(explicit_datasets.keys()) - filter(is_parameter, pipeline_datasets)):
```

and for me, it looks way more complex to understand than

```
for ds_name in pipeline_datasets:
    if ds_name in explicit_datasets or is_parameter(ds_name):
        continue
```

It's also slightly worse from the complexity point of view though I agree both cost nothing.

Curious what other think.